mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/CheckCmd.scala

```diff
@@ -50,4 +59,75 @@ class CheckCmd(name: String = "check", description: String = "Check a TLA+ speci
   var view: String =
     opt[String](name = "view", description = "the state view to use with --max-error=n, default: transition index",
         default = "")
+
+  def setTuningOptions(): Map[String, String] = {
+    val tuning =
+      if (tuningOptionsFile != "") loadProperties(tuningOptionsFile) else Map[String, String]()
+    overrideProperties(tuning, tuningOptions)
+  }
+
+  def run() = {
+
+    val tuning = setTuningOptions()
+
+    logger.info("Tuning: " + tuning.toList.map { case (k, v) => s"$k=$v" }.mkString(":"))
+
+    executor.passOptions.set("general.tuning", tuning)
+    executor.passOptions.set("checker.nworkers", nworkers)
+    executor.passOptions.set("checker.discardDisabled", discardDisabled)
+    executor.passOptions.set("checker.noDeadlocks", noDeadlocks)
+    executor.passOptions.set("checker.algo", algo)
+    executor.passOptions.set("checker.smt-encoding", smtEncoding)
+    executor.passOptions.set("checker.maxError", maxError)
+    if (view != "")
+      executor.passOptions.set("checker.view", view)
+    // for now, enable polymorphic types. We probably want to disable this option for the type checker
+    executor.passOptions.set("typechecker.inferPoly", true)
+    setCommonOptions()
```

Comment:
Yeah, I think this is isn't really relevant now, but mostly just because it's easy now to reason about how the order these things are set, so the reader can determine which order they want to set things.